### PR TITLE
fix(provider/google): Handling edge cases with user data files better.

### PR DESCRIPTION
**Pull request by varikin**
This incorporates 3 changes/improvements for handling user data files. It
allows blank lines in the user data file. It also interprets the lack of an
equal sign as a key with no value. Finally, it allows values to contain equal
signs, in other words, it only splits on the first equal sign.

### CODE REVIEW

**Comment by varikin** [comment]
@dbakshiyev @duftler here is my update to the user data file stuff. I hope my commit/pr message has enough context. The major issue was that blank lines in the udf file were turned into key/value pairs of empty strings which GCE does not like. Beyond that, I cleaned up some edge cases and refactored some things so I could test the filtering more easily.

**Comment by varikin** [comment]
Hmm...And I meant to make the type in the commit message by fix, not feature.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy [hunk 1]
```diff
@@ -375,8 +375,7 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
     }
 
     List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicy =
-      description.autoHealingPolicy?.healthCheck
-      ? [new InstanceGroupManagerAutoHealingPolicy(
+      description.autoHealingPolicy?.healthCheck ? [new InstanceGroupManagerAutoHealingPolicy(
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy]: Is this reformatting intentional? Don't see a code change for this line.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProvider.groovy [hunk 2]
```diff
@@ -92,40 +101,45 @@ public class GoogleUserDataProvider {
...
-      .replace('%%region%%', description.region ?: '')
-      .replace('%%stack%%', description.stack ?: '')
-      .replace('%%group%%', names.group ?: '')
-      .replace('%%cluster%%', names.cluster ?: '')
-      .replace('%%detail%%', names.detail ?: '')
-      .replace('%%launchconfig%%', instanceTemplateName ?: '')
-
-    return result
+    return rawUserData.collect{userData ->
+      return userData.replace('%%account%%', description.accountName)
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProvider.groovy]: I think it reads slightly better with all of the `.replace()`'s aligned.

**Comment by duftler** [comment]
Could also update the spinnaker [readme](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/UserData.md) on userdata to describe the edge cases mentioned in this pr. Seems like that shouldn't exist just under clouddriver-aws, so we should probably move that elsewhere.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy (hunk 1 above)
Comment (varikin) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy]: Intellij was complaining about the `?` on the second line

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProvider.groovy (hunk 2 above)
Comment (varikin) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProvider.groovy]: I can do that.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy (hunk 1 above)
Comment (duftler) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy]: Yeah. The folks here make fun of me for formatting it this way. I prefer the original style, but can live with it either way. I do typically try to avoid reformatting lines that don't have code changes unless it really affects readability.

**Comment by duftler** [comment]
Thanks for digging into this.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy (hunk 1 above)
Comment (varikin) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy]: I can adjust it. I might have some intellij setting that was flagging it as an error and so wouldn't compile.

**Comment by varikin** [comment]
@duftler would you prefer updating the readme or updating spinnaker.io documentation? I added a section when doing the [Openstack work to include it](https://www.spinnaker.io/setup/providers/openstack/#common-userdata). I can update and move that to a common area in the docs if you would prefer that.

**Comment by duftler** [comment]
@varikin I think the best approach would be to do what you proposed: update the spinnaker.io/openstack/userdata section and move it to a common area (and probably list what providers have support for this feature).

**Comment by duftler** [comment]
Going to cherrypick this commit into a new branch so I can merge it without you having to rebase again.

**Comment by duftler** [comment]
Probably also want to add some text describing the 3 edge cases covered in this pr.

**Comment by duftler** [comment]
Closing this PR since the commit was cherrypicked in #1852.
